packages/new-block-stream/fromPolling.js

```diff
@@ -9,21 +9,20 @@ const fromPolling = ({ web3, pollingInterval }) => {
   const provider = web3.currentProvider;
   const blockTracker = new PollingBlockTracker({ provider, pollingInterval });
 
-  const observable = new Observable(subscriber => {
-    blockTracker
-      .on("latest", async blockNum => {
-        // get full block info with `web3.eth.getBlock`
-        const block = await web3.eth.getBlock(blockNum, true);
-        subscriber.next(block);
-      })
-      .on("error", err => subscriber.next(err));
-  });
+  const observable = new Subject();
```

Comment:
What's the reason for changing to Subject? I seem to recall an earlier discussion that we had on changing the Subjects to Observables, since you couldn't push your own events to an Observable it was safer.

Reply:
You need the Subject to be able to call `.complete` on it outside of the function. I suppose you could assign it to a temp variable by using its closure.